packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -97,20 +97,20 @@ export const Product = ({
     selectedPlan = productPlans[0];
   }
 
-  const onPayment = useCallback((tokenResponse: stripe.TokenResponse) => {
+  const onPayment = useCallback((tokenResponse: stripe.TokenResponse, name: string) => {
     if (tokenResponse && tokenResponse.token) {
       createSubscription(accessToken, {
         paymentToken: tokenResponse.token.id,
         planId: selectedPlan.plan_id,
-        displayName: profile.result ? profile.result.displayName : '',
+        displayName: name,
       });  
     } else {
       // This shouldn't happen with a successful createToken() call, but let's
       // display an error in case it does.
       const error: any = { message: 'No token response received from Stripe' };
       setCreateTokenError(error);
     }
-  }, [ accessToken, selectedPlan, createSubscription, setCreateTokenError ]);
+  }, [ accessToken, name, selectedPlan, createSubscription, setCreateTokenError ]);
```

Comment:
You shouldn't need to add `name` to the list of closure-captured dependencies here, since it gets passed in as a param to the callback being memoized by the hook.

Reply:
Thanks, yeah, didn't initially realize this was a list of implied globals